test/producer-consumer-spec.js

```diff
@@ -1,183 +1,199 @@
 };
 
 var letters = 0;
 
-describe('Producer/Consumer msg delevering:', function() {
-
-  it('should be able to consume message sended by producer to queue [test-queue-0]', function (done) {
-    producer.produce(fixtures.queues[0], { msg: uuid.v4() })
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(typeof _msg === 'object');
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
+describe('producer/consumer', function() {
+  before(function() {
+    this.timeout(0);
+    return docker.start();
   });
 
-  it('should be able to consume message sended by producer to queue [test-queue-0] (no message)', function (done) {
-    producer.produce(fixtures.queues[0])
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(_msg === undefined);
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
+  after(() => {
+    return docker.stop();
   });
 
-  it('should be able to consume message sended by producer to queue [test-queue-0] (null message)', function (done) {
-    producer.produce(fixtures.queues[0], null)
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(_msg === null);
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
-  });
+  describe('msg delevering', function() {
 
-  it('should not be able to consume message sended by producer to queue [test-queue-1]', function (done) {
-    producer.produce(fixtures.queues[1], { msg: uuid.v4() })
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(typeof _msg === 'object');
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
-  });
+    it('should be able to consume message sended by producer to queue [test-queue-0]', function (done) {
+      producer.produce(fixtures.queues[0], { msg: uuid.v4() })
+      .then(function (response) {
+        assert(response === true);
+        ++letters;
+      })
+      .then(function () {
+        return consumer.consume(fixtures.queues[0], function (_msg) {
+          assert(typeof _msg === 'object');
+        });
+      })
+      .then(function (response) {
+        assert(response === true);
+        --letters;
+      })
+      .then(done);
```

Comment:
Yeah, but I have to rewrite it just noticed its badly written. I want to resolve when response is actually received, done() is not well placed (it should be in a callback of consume method). Will fix that